internal/graph/checkdispatchset_test.go

```diff
@@ -0,0 +1,387 @@
+				"group:3#member": {
+					{resourceID: "somedoc", caveat: nil},
+				},
+			},
+		},
+		{
+			"different resources leading to the same subject",
+			[]*core.RelationTuple{
+				tuple.MustParse("document:somedoc#viewer@group:1#member"),
+				tuple.MustParse("document:anotherdoc#viewer@group:1#member"),
+				tuple.MustParse("document:somedoc#viewer@group:2#member"),
+				tuple.MustParse("document:anotherdoc#viewer@group:2#member"),
+			},
+			2,
+			[]checkDispatchChunk{
+				{
+					resourceType:       relationRef{namespace: "group", relation: "member"},
+					resourceIds:        []string{"1", "2"},
+					hasIncomingCaveats: false,
+				},
+			},
+			map[string][]resourceIDAndCaveat{
+				"group:1#member": {
+					{resourceID: "somedoc", caveat: nil},
+					{resourceID: "anotherdoc", caveat: nil},
+				},
+				"group:2#member": {
+					{resourceID: "somedoc", caveat: nil},
+					{resourceID: "anotherdoc", caveat: nil},
+				},
+			},
+		},
+		{
+			"different resources leading to the same subject with caveats",
+			[]*core.RelationTuple{
+				tuple.MustParse("document:somedoc#viewer@group:1#member[somecaveat]"),
+				tuple.MustParse("document:anotherdoc#viewer@group:1#member"),
+				tuple.MustParse("document:somedoc#viewer@group:2#member"),
+				tuple.MustParse("document:anotherdoc#viewer@group:2#member[somecaveat]"),
+			},
+			2,
+			[]checkDispatchChunk{
+				{
+					resourceType:       relationRef{namespace: "group", relation: "member"},
+					resourceIds:        []string{"1", "2"},
+					hasIncomingCaveats: true,
+				},
+			},
+			map[string][]resourceIDAndCaveat{
+				"group:1#member": {
+					{resourceID: "somedoc", caveat: caveatForTesting("somecaveat")},
+					{resourceID: "anotherdoc", caveat: nil},
+				},
+				"group:2#member": {
+					{resourceID: "somedoc", caveat: nil},
+					{resourceID: "anotherdoc", caveat: caveatForTesting("somecaveat")},
+				},
+			},
+		},
+		{
+			"different resource leading to the same subject with caveats",
+			[]*core.RelationTuple{
+				tuple.MustParse("document:anotherdoc#viewer@group:1#member"),
+				tuple.MustParse("document:thirddoc#viewer@group:1#member"),
+				tuple.MustParse("document:somedoc#viewer@group:1#member[somecaveat]"),
+			},
+			2,
+			[]checkDispatchChunk{
+				{
+					resourceType:       relationRef{namespace: "group", relation: "member"},
+					resourceIds:        []string{"1"},
+					hasIncomingCaveats: true,
+				},
+			},
+			map[string][]resourceIDAndCaveat{
+				"group:1#member": {
+					{resourceID: "somedoc", caveat: caveatForTesting("somecaveat")},
+					{resourceID: "anotherdoc", caveat: nil},
+					{resourceID: "thirddoc", caveat: nil},
+				},
+			},
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			set := newCheckDispatchSet()
+			for _, rel := range tc.relationships {
+				set.addForRelationship(rel)
+			}
+
+			chunks := set.dispatchChunks(tc.dispatchChunkSize)
+			for _, c := range chunks {
+				sort.Strings(c.resourceIds)
+			}
```

Comment:
why sorting, isn't this already supposed to be sorted by `dispatchChunks`?

Reply:
Was added before I had the call sorting it